packages/storage-plus/src/de.rs

```diff
@@ -0,0 +1,119 @@
+use crate::keys::{IntKey, TimestampKey};
+use cosmwasm_std::{Addr, StdError, StdResult};
+use std::array::TryFromSliceError;
+use std::convert::TryInto;
+
+pub trait KeyDeserialize {
+    type Output: Sized;
+
+    fn from_slice(value: &[u8]) -> StdResult<Self::Output>;
+}
+
+impl KeyDeserialize for () {
+    type Output = ();
+
+    fn from_slice(_value: &[u8]) -> StdResult<Self::Output> {
+        Ok(())
+    }
+}
+
+macro_rules! bytes_de {
```

Comment:
I am really not happy with this kind of macro. It saves you couple keystrokes, but at the end makes code way less readable (including strange error messages in some cases). Why not to implement it just for `&[u8]` and if you really need implementation for another 3 types forward it (so in case of changes, there is only one place to change)? Same for `string_de`.

Reply:
Good idea.